module/extension/Director/src/Director.cpp

```diff
@@ -262,8 +263,8 @@ namespace module::extension {
         // the Provider has finished executing
         on<Trigger<BehaviourTask>>().then("Director Task", [this](const BehaviourTask& t) {
             std::lock_guard<std::recursive_mutex> lock(director_mutex);
-            // Make our own mutable copy of the task
-            auto task = std::make_shared<BehaviourTask>(t);
+            // Make our own mutable director task from the behaviour task
+            auto task = std::make_shared<DirectorTask>(t, false);
```

Comment:
I would just make false be a default argument on the DirectorTask constructor rather than passing it in